contracts/contracts/MACI.sol

```diff
@@ -129,6 +131,7 @@ contract MACI is IMACI, DomainObjs, Params, SnarkCommon, Ownable {
         pollFactory = _pollFactory;
         signUpGatekeeper = _signUpGatekeeper;
         initialVoiceCreditProxy = _initialVoiceCreditProxy;
+        stateTreeDepth = _stateTreeDepth;
```

Comment:
Is there any danger in this value now being an arbitrary size? i.e. should there be a check against some max value?

Reply:
Yes and no. If they put a value that does not match the circuit, they would basically need to start over (there might be some workarounds though, still not ideal).

This is a uint8 so max value would be 255, having said that, a tree depth of 255 would be incredibly big, considering 10 allows for a little bit more than 1M total signups. 
My view here is the following:
1. if we put a max value, what should this be?
2. In the past, we asked the users to change a constant variable rather than passing it as a constructor parameter:
   - I consider these options kinda the same, we never really validated that this constant variable was > than a max value
   - people running MACI should be (and are to some extent) informed that this parameter should match the one in the circuits, hence it is their responsibility of setting it up correctly.
3. even if this value was < than a max value we decide on, it could still be different than what they put in the circuit and in the other components of MACI. 

To summarize, I think the best we can do here in terms of "validation" is to properly inform that this value should match whichever value is in the Circom circuits.